edward/inferences/klqp.py

```diff
@@ -92,25 +93,25 @@ def build_loss_and_gradients(self, var_list):
 
     of the loss function.
 
-    If the variational model is a normal distribution and the prior is
-    standard normal, then loss function can be written as
+    If the KL divergence between the variational model and the prior
+    is tractable, then the loss function can be written as
 
     .. math::
 
       -\mathbb{E}_{q(z; \lambda)}[\log p(x \mid z)] +
         \\text{KL}( q(z; \lambda) \| p(z) ),
 
     where the KL term is computed analytically (Kingma and Welling,
-    2014).
+    2014). For example, this is true when :math:`p(z)` and :math:`q(z;
+    \lambda)` are Normal.
     """
     is_reparameterizable = all([rv.is_reparameterized and rv.is_continuous
                                 for rv in six.itervalues(self.latent_vars)])
-    qz_is_normal = all([isinstance(rv, Normal) for
-                       rv in six.itervalues(self.latent_vars)])
-    z_is_normal = all([isinstance(rv, Normal) for
-                       rv in six.iterkeys(self.latent_vars)])
-    is_analytic_kl = qz_is_normal and \
-        (z_is_normal or hasattr(self.model_wrapper, 'log_lik'))
+    # TODO(dt): determine if there is a kl registered for kl(qz, z)
+    # for each pair in self.latent_vars. how?
```

Comment:
Looks like the best you can do for now is catch `NotImplementedError` when calling `distributions.kl`. I guess you could make a helper function to do the function call, but it's pretty awkward and will help some junk in the graph. This is probably something worth adding to TensorFlow. CC @ebrevdo